src/microbe_stage/systems/MicrobeAISystem.cs

```diff
@@ -1320,34 +1419,45 @@ private void BuildChunksCache()
         // To allow multithreaded AI access safely
         lock (chunkDataCache)
         {
-            if (chunkCacheBuilt)
-                return;
-
-            foreach (ref readonly var chunk in chunksSet.GetEntities())
+            lock (terrainChunkDataCache)
```

Comment:
I think this second level of locking is unnecessary and could be replaced with a comment like:
```
// As the chunk lock is always held when building all of the chunk caches, the other individual cache objects don't need separate locks to protect them
```